### PR TITLE
build: move to PEP517 and pyproject.toml, drop support for Py3.6 and Py3.7

### PR DESCRIPTION
**Pull request by GabDug**
Hey :wave: 

Following the recent packaging changes, I propose to go a step further and move to pyproject.toml, to use more modern Python packaging tooling.

This change will enable us to use any modern build-system, like recent `setuptools`, `pdm-backend` `hatchling`, which supports plugins. This could be useful to set the version automatically from the Git tag for example.

Unfortunately, Python 3.6 being no longer supported, I had to drop its support. I opened an issue #418 to discuss Python support policy.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -817,7 +815,6 @@ pytest
 The library is also tested across many versions of Python and Django. To run tests that way:
 
 ```sh
-pyenv install -s 3.6
 pyenv install -s 3.7
```
Comment (EmilStenstrom) [inline README.md:818]: 3.7 should be dropped too right?

File: pyproject.toml [hunk 2]
```diff
@@ -1,3 +1,48 @@
...
+    "Framework :: Django :: 4.2",
+    "Framework :: Django :: 5.0",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
```
Comment (EmilStenstrom) [inline pyproject.toml:29]: Slightly strange ordering of trove classifiers?

File: setup.py [hunk 3]
```diff
@@ -1,43 +1,6 @@
-# -*- coding: utf-8 -*-
-import os
+# Configuration has moved to pyproject.toml
+# This file is maintained for backward compatibility
```
Comment (EmilStenstrom) [inline setup.py:2]: What kinds of setup is this supporting? Why can't we remove this file entirely?

File: scripts/supported_versions.py [hunk 4]
```diff
@@ -169,7 +192,7 @@ def build_pyenv(python_to_django: VersionMapping):
 
 
 def build_ci_python_versions(python_to_django: Dict[str, str]):
-    # Outputs python-version: ['3.6', '3.7', '3.8', '3.9', '3.10', '3.11']
+    # Outputs python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
```
Comment (EmilStenstrom) [inline scripts/supported_versions.py:195]: 3.7 should be remove too right?

**Review by EmilStenstrom — COMMENTED** [review]
Found a couple of simple fixes, but looks good! Happy to be moving towards modern python packaging. Does this change how to release the package?

File: pyproject.toml (hunk 2 above)
Comment (GabDug) [reply pyproject.toml:29]: You're right, probably due to an automatic sorting on import. Thanks!

**Comment by GabDug** [comment]
> Does this change how to release the package?

This does not, if you where using `python -m build` or any other PEP517 aware tool.

File: setup.py (hunk 3 above)
Comment (GabDug) [reply setup.py:2]: You're probably right, I think we could remove the file entirely. 

Only some old versions of pip/setuptools/etc might not support `pyproject.toml` projects correctly, but on 3.8+ users are advised to upgrade.

This should not be an issue, as we distribute universal wheels: almost no users would package from the sdist.

**Comment by EmilStenstrom** [comment]
@GabDug The build process is fully automated, and is currently based on a GitHub Action that reacts of new tags being pushed. So to release I (or anyone else with commit permissions to this repo) just update the version number, tag a commit with that version number and push the tag to the repo. 

https://github.com/EmilStenstrom/django-components/blob/master/.github/workflows/publish-to-pypi.yml

**Comment by GabDug** [comment]
> @GabDug The build process is fully automated, and is currently based on a GitHub Action that reacts of new tags being pushed. So to release I (or anyone else with commit permissions to this repo) just update the version number, tag a commit with that version number and push the tag to the repo.
> 
> https://github.com/EmilStenstrom/django-components/blob/master/.github/workflows/publish-to-pypi.yml

This workflow uses `python -m build` so should work as is. Only change is you now have to edit the version in `pyproject.toml`.

I haven't setup `setuptools.scm` to retrieve the version automatically from the Git tag, I think it can come in another PR.

**Pull request merged**